src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala

```diff
@@ -498,22 +510,30 @@ class EthService(
     * @return The syncing status if the node is syncing or None if not
     */
  def syncing(req: SyncingRequest): ServiceResponse[SyncingResponse] = Future {
-   val currentBlock = appStateStorage.getBestBlockNumber()
-   val highestBlock = appStateStorage.getEstimatedHighestBlock()
-
-   //The node is syncing if there's any block that other peers have and this peer doesn't
    val maybeSyncStatus =
-     if(currentBlock < highestBlock)
+     if(isSyncing)
        Some(SyncingStatus(
          startingBlock = appStateStorage.getSyncStartingBlock(),
-         currentBlock = currentBlock,
-         highestBlock = highestBlock
+         currentBlock = appStateStorage.getBestBlockNumber(),
+         highestBlock = appStateStorage.getEstimatedHighestBlock()
        ))
      else
        None
    Right(SyncingResponse(maybeSyncStatus))
  }
 
+  /**
+    * Returns whether the client is syncing
+    * The node is assumed to be syncing only if there's any block that other peers have and this peer doesn't
+    *
+    * @return if the node is syncing
+    */
+  private def isSyncing: Boolean = {
```

Comment:
So...this will only mine if we have all the blocks the latest block we know (for example defined in `io.iohk.ethereum.network.EtcPeerManagerActor#handleReceivedMessage`), right?

Could it happen that another dishonest peer send us a really big block to prevent us from mining?

Reply:
It seems that could be the case with the current implementation (Parity seems to have ways of preventing that), changing the state of the PR to WIP to look into handling this.